packages/flutter/lib/src/physics/spring_simulation.dart

```diff
@@ -78,6 +116,43 @@ class SpringDescription {
   /// driving the [SpringSimulation].
   final double damping;
 
+  /// The duration parameter used in [SpringDescription.withDurationAndBounce].
+  ///
+  /// This value defines the perceptual duration of the spring, controlling
+  /// its overall pace. It is approximately equal to the time it takes for
+  /// the spring to settle, but for highly bouncy springs, it instead
+  /// corresponds to the oscillation period.
+  ///
+  /// This duration does not represent the exact time for the spring to stop
+  /// moving. For example, when [bounce] is `1`, the spring oscillates
+  /// indefinitely, even though [duration] has a finite value. To determine
+  /// when the motion has effectively stopped within a certain tolerance,
+  /// use [SpringSimulation.isDone].
+  ///
+  /// Defaults to 0.5 seconds.
+  Duration get duration {
+    final double durationInSeconds = math.sqrt((4 * math.pi * math.pi * mass) / stiffness);
+    final int milliseconds = (durationInSeconds * Duration.millisecondsPerSecond).round();
+    return Duration(milliseconds: milliseconds);
+  }
+
+  /// The bounce parameter used in [SpringDescription.withDurationAndBounce].
+  ///
+  /// This value controls how bouncy the spring is:
+  ///
+  ///  * A value of 0 results in a critically damped spring with no oscillation.
+  ///  * Values between 0 and 1 produce underdamping, where the spring oscillates a few times
+  ///    before settling. A value of 1 represents an undamped spring that
+  ///    oscillates indefinitely.
+  ///  * Negative values indicate overdamping, where the motion is slow and
+  ///    resistive, like moving through a thick fluid. The minimum value is -1.0.
```

Comment:
Now that it has no caps:
```suggestion
  ///  * Negative values indicate overdamping, where the motion is slow and
  ///    resistive, like moving through a thick fluid.
```